### PR TITLE
Add unit conversion tests

### PR DESCRIPTION
**Pull request by javerbukh**
Add more comprehensive unit conversion tests that also test the uncertainty.

### CODE REVIEW

File: jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py [hunk 1]
```diff
@@ -1,92 +1,278 @@
...
 import astropy.units as u
 import numpy as np
-from specutils.spectra import Spectrum1D
-
+from specutils import Spectrum1D, SpectrumCollection
+from jdaviz.configs.specviz.helper import SpecViz
 from .. import unit_conversion as uc
+import pytest
+from astropy.nddata import VarianceUncertainty, StdDevUncertainty,\
+    InverseVariance, UnknownUncertainty
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py:9]: Slash has been discouraged for a while. Use parenthesis.
```suggestion
from astropy.nddata import (VarianceUncertainty, StdDevUncertainty,
                            InverseVariance, UnknownUncertainty)
```

**Review by pllim — COMMENTED** [review]
If you are re-using the same fixture over and over again (I think I see these in your other PR), might want to consider defining it in an upper level, like in `conftest.py` somewhere.

Look into https://docs.pytest.org/en/6.2.x/parametrize.html for tests that seem to repeat the same code.

File: jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py (hunk 1 above)
Comment (javerbukh) [reply jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py]: Did not know that, thanks!

**Comment by javerbukh** [comment]
Got it! I moved the fixtures to conftest. I will prioritize getting this PR merged so I can just rebase my other two.

**Comment by pllim** [comment]
The rest of my review comments are over at javerbukh/jdaviz#6 .

**Review by pllim — APPROVED** [review]

**Review by rosteen — COMMENTED** [review]
@javerbukh you said in the description that these tests "also test the uncertainty," but I don't see any `assert` statements that relate to the uncertainty. Are you just testing that the unit conversion doesn't break if there's an uncertainty on the spectrum, or did you really want to test that the uncertainty units change as well?

**Comment by javerbukh** [comment]
Mostly the former, although it's a good idea to test the latter as well. I'll work on that now.

File: jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py [hunk 2]
```diff
@@ -1,92 +1,247 @@
...
+    new_flux = "erg / (s cm2 um)"
+
+    conv_func = uc.UnitConversion.process_unit_conversion
+    converted_spectrum = conv_func(specviz_app.app, spectrum=spectrum1d,
+                                   new_flux=new_flux)
+
+    assert_quantity_allclose(converted_spectrum.flux,
+                             RESULT_FLUX, atol=1e-5*u.Unit(new_flux))
+    assert_quantity_allclose(converted_spectrum.uncertainty.quantity.value,
+                             RESULT_UNCERTAINTY, atol=1e-5)
```
Comment (rosteen) [inline jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py:129]:
```suggestion
                             RESULT_UNCERTAINTY, atol=1e-11)
```
The absolute tolerance value needs to be smaller, given that the values being compared are ~e-9

File: jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py [hunk 3]
```diff
@@ -1,92 +1,247 @@
...
-    assert np.allclose(converted_spectrum.flux.value[1:], result_flux,
-                       atol=1e-5)
-    assert np.allclose(converted_spectrum.spectral_axis.value,
-                       result_spectral_axis, atol=1e-5)
+    assert_quantity_allclose(converted_spectrum.spectral_axis,
+                             RESULT_SPECTRAL_AXIS, atol=1e-5*u.um)
+    assert_quantity_allclose(converted_spectrum.flux,
+                             RESULT_FLUX, atol=1e-5*u.Unit(new_flux))
+    assert_quantity_allclose(converted_spectrum.uncertainty.quantity.value,
+                             RESULT_UNCERTAINTY, atol=1e-5)
```
Comment (rosteen) [inline jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py:247]:
```suggestion
                             RESULT_UNCERTAINTY, atol=1e-11)
```

**Review by rosteen — CHANGES_REQUESTED** [review]

File: jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py (hunk 2 above)
Comment (javerbukh) [reply jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py]: Good catch!

**Review by rosteen — APPROVED** [review]

**Comment by rosteen** [comment]
Looks good to me now, merging.

**Pull request merged**